homeassistant/components/upnp/sensor.py

```diff
@@ -135,7 +167,11 @@ def native_value(self) -> str | None:
         value = self.coordinator.data[device_value_key]
         if value is None:
             return None
-        return format(value, "d")
+        return (
+            format(value, self._sensor_type["format"])
+            if self._sensor_type.get("format")
```

Comment:
I wouldn't mind adding `s` as format for `ROUTER_IP` and `WAN_STATUS`. This way, there is no need to check if the sensor type has a `format` entry, simplifying code a bit.